feat(auth): add user authentication

mohamed952741 opened the pull request:
- Implemented static user authentication using an array of JSON Objs
- Validate user login and display error message
- Store login details in localStorage and redirect to index.html on success

Closes #12

mohamed952741 committed:
feat(auth): add user authentication
- Implemented static user authentication using an array of JSON Objs
- Validate user login and display error message
- Store login details in localStorage and redirect to index.html on success

Closes #12

AAEmara commented on scripts/auth.js:
```diff
@@ -0,0 +1,45 @@
+    { username: "Hussein", password: "Hussein" }
+];
+
+//add event listner when page is loaded
```
```suggestion
//add event listener when page is loaded
```
A little typo in the comment.

AAEmara commented on scripts/auth.js:
```diff
@@ -0,0 +1,45 @@
+    
+    //get HTML elements
+    const loginBtn = document.getElementById("loginBtn");
+    const errorText = document.getElementById("WrongPass");
```
```suggestion
    const errorText = document.getElementById("wrongPass");
```
Please consider changing the element's ID, since it was merged as "wrongPass" in issue #7.

AAEmara reviewed (changes requested):
Please merge the changes added to the main branch and follow the requested changes.

mohamed952741 committed:
fix(auth): resolve merge conflicts
- Merged remote-tracking branch 'origin/main' into feat/login-flow
- Fixed typo on lines 11, 21 in file auth.js
- Changing the element's ID "WrongPass" -> "wrongPass"
Closes #12

AAEmara committed:
Merge branch 'main' into feat/login-flow

AAEmara commented:
Great Job, Mohamed! Everything works well, but we had some tweaks to resolve the conflicts with #14 issue.
